src/URI.js

```diff
@@ -487,6 +487,36 @@
 
   URI.encodeReserved = generateAccessor('reserved', 'encode');
 
+  URI.toUri = function(string, defaults) {
+    /**
+     * Convert a string which may or may not have a valid scheme
+     * to a valid URI prepending a scheme if needed
+     * @param {string} string   The url to parse
+     * @param {object} defaults The defaults to use for the URI properties if not parsed.
+     *        Currently only supports, defaults.scheme which should be one of [https, http]
+     *        '//' will be added if no scheme is parsed or provided
+     * @return {string} A valid URI
+     *
+     */
+    defaults = defaults || {};
+    // if the uri doesn't have a scheme, add one
+    // now so it doesn't get parsed as a path
+    if (!string.match(/^(https|http)?:\/\//i)) {
+      // add the given scheme if its a valid one
+      switch (true) {
+        case defaults.scheme === 'http':
+        case defaults.scheme === 'https':
+          defaults.scheme = defaults.scheme+':';
```

Comment:
you're mutating a an object that was passed into the function, this can have unexpected consequences. consider a simplified version of your code:

```js
function doSomething(defaults) {
  switch (true) {
    case defaults.scheme === 'http':
    case defaults.scheme === 'https':
      defaults.scheme = defaults.scheme+':';
      break;
    default:
      defaults.scheme = '';
  }

  console.log(defaults)
}

var myDefaults = { scheme: 'http' };
doSomething(myDefaults);
doSomething(myDefaults);
```

and the output:

```
{scheme: "http:"}
{scheme: ""}
```